elasticsearch/repository.sls

```diff
@@ -1,18 +1,35 @@
-{% from "elasticsearch/map.jinja" import elasticsearch with context %}
-{% set os_family = grains['os_family'] %}
+{% from "elasticsearch/map.jinja" import elasticsearch, elasticsearch_repo with context %}
+{% set os_family = salt.grains.get('os_family') %}
+{% set osfullname = salt.grains.get('osfullname') %}
 
 {% if os_family == 'RedHat' %}
 install_elasticsearch_gpg_key:
   cmd.run:
-    - name: rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch
+    - name: rpm --import {{ elasticsearch.gpg-key }}
     - require_in:
         - pkgrepo: configure_elasticsearch_package_repo
 {% endif %}
 
+{% if elasticsearch.elastic_stack %}
+configure_elasticsearch_package_repo:
+  pkgrepo.managed:
+    - humanname: 'Elasticsearch 5.x'
+    {% if os_family == 'Debian' %}
+    - name: deb {{ elasticsearch_repo.pkg_repo_url }}/apt stable main
+    - gpgkey: {{ elasticsearch.gpg_key }}
+    - refresh_db: True
+    {% elif os_family == 'RedHat' %}
+    - name: {{ name }}
+    - baseurl: {{ elasticsearch_repo.pkg_repo_url }}/yum
+    - gpgcheck: 1
+    - enabled: 1
+    {% endif %}
+    - key_url: {{ elasticsearch.gpg_key }}
+{% else %}
 {% for name, version in elasticsearch.products.items() %}
-configure_{{ name }}_package_repo:
+configure_elasticsearch_package_repo:
```

Comment:
This line needs to stay the same since it is a for loop and will otherwise result in a duplicate state id error.